hvplot/converter.py

```diff
@@ -819,6 +826,17 @@ def _process_data(self, kind, data, x, y, by, groupby, row, col,
             data, x, y, by_new, groupby_new = process_xarray(
                 data, x, y, by, groupby, use_dask, persist, gridded,
                 label, value_label, other_dims, kind=kind)
+            if robust:
+                # adapted from xarray
+                # https://github.com/pydata/xarray/blob/main/xarray/plot/utils.py#L729
+                vmax = np.nanpercentile(data[z], 100 - 2)
+                vmin = np.nanpercentile(data[z], 2)
+                # Scale interval [vmin .. vmax] to [0 .. 1], with darray as 64-bit float
+                # to avoid precision loss, integer over/underflow, etc with extreme inputs.
+                # After scaling, cast to 8-bit integer.
+                data[z] = (
+                    ((data[z].astype("f8") - vmin) / (vmax - vmin)) * 255
+                ).astype(int)
```

Comment:
I was a bit surprised to see this warning running this example with `robust=True`:

> WARNING:param.RGBPlot00947: Clipping input data to the valid range for RGB data ([0..1] for floats or [0..255] for integers).

It seems to me the different with xarray's implementation is that they clip the data to [0..1]

https://github.com/pydata/xarray/blob/6af547cdd9beac3b18420ccb204f801603e11519/xarray/plot/utils.py#L759

Why do we deviate from xarray's implementation? There may be a good reason I'm missing! If there is, perhaps the docstring shouldn't say the data is clipped?

Reply:
The clipping is happening in HoloViews if that's what you're asking.

Reply:
But this leads to HoloViews displaying this warning. Do you expect that?

> WARNING:param.RGBPlot00947: Clipping input data to the valid range for RGB data ([0..1] for floats or [0..255] for integers).